includes/api/ApiQueryUserContributions.php

```diff
@@ -486,4 +498,42 @@ public function getHelpUrls() {
 	public function getVersion() {
 		return __CLASS__ . ': $Id$';
 	}
+
+	private function addWhereOr( $conditions ) {
+		$where = [];
+		foreach ( $conditions as $name => $values ) {
+			if (is_array( $values ) && count($values) === 1) {
+				$values = $values[0];
+			}
+
+			$isEmptyArray = is_array( $values ) && count( $values ) === 0;
+			if ( $values && !$isEmptyArray ) {
+				$where[] = $this->createWhereCondition( $name, $values );
+			}
+		}
+
+		$this->addWhere( '(' . implode( ' OR ', $where ) . ')' );
+	}
+
+	private function normalizeValues( $values ) {
+		return array_map( function ($value) {
+			if (is_numeric($value)) {
+				return $value;
+			}
+
+			return "'$value'";
+		}, $values );
+	}
+
+	private function createWhereCondition( $name, $values ): string {
```

Comment:
`Database::makeList` looks pretty similar to this custom method. Can we rely on the existing functionality?